### PR TITLE
Feature/21

### PR DESCRIPTION
**Pull request by isauros1000**
First attempt at the Post Mutation!

### CODE REVIEW

File: server/graphql/resolvers/listing.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const { AuthenticationError } = require('apollo-server');
+
+const {Listing} = require('../../models/Listing');
+
+module.exports = {
+  Mutation: {
+    createDog: async (_, { title, description, price, active=true, pictures, pickup, category }, context) => {
```
Comment (DanielCufino) [inline server/graphql/resolvers/listing.js:7]: This should be createListing

File: server/graphql/resolvers/listing.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
```
Comment (DanielCufino) [inline server/graphql/resolvers/listing.js:8]: In the future we're going to want to add authentication handling. I.E. Authenticating whether the using logged in is the user trying to create this listing. This is fine for now though it's not a huge priority

File: server/models/Listing.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
 const { model, Schema } = require('mongoose');
-const Image = require('./Image.js')
+const Image = require('./Image.js').schema;
```
Comment (DanielCufino) [inline server/models/Listing.js:2]: This may be the issue that you're running into. Try removing the .schema

File: server/models/Listing.js [hunk 4]
```diff
@@ -17,9 +17,10 @@ const listingSchema = new Schema({
   active: { type: Boolean, required: true },
   pictures: [Image],
```
Comment (DanielCufino) [inline server/models/Listing.js:18]: On second thought, we probably shouldn't refer to entire Image schema objects within a listing but instead the object ID's of the images.

Try this instead:

`pictures: [{ type: Schema.Types.ObjectId, ref: 'Image' }]`

File: server/models/Listing.js [hunk 5]
```diff
@@ -17,9 +17,10 @@ const listingSchema = new Schema({
...
   pickup: Boolean,
-  category: categories
+  category: { type: String, enum: ['CLOTHES', 'FURNITURE', 'TECHNOLOGY', 'JEWELRY', 'SERVICE', 'OTHER']}
 });
 
 
 const Listing = model('Listing', listingSchema);
-module.exports = Listing;
```
Comment (DanielCufino) [inline server/models/Listing.js:25]: I would shy away from this and just leave it as a one-liner:

`module.exports = model('Listing', listingSchema);`

Comment (DanielCufino) [reply server/models/Listing.js:25]: I think there are multiple occurences of this in the PR

**Review by DanielCufino — CHANGES_REQUESTED** [review]
Looking good! A couple of nits in the comments

File: server/models/Listing.js (hunk 4 above)
Comment (isauros1000) [reply server/models/Listing.js:18]: good idea! I had seen this in a stack exchange when I was debugging but I wasn't quite sure that it meant but thank you for explaining.

File: server/graphql/resolvers/listing.js (hunk 2 above)
Comment (isauros1000) [reply server/graphql/resolvers/listing.js:8]: In the future, how would we go about this ? Add a user parameter and look up to see if they're logged in?

Comment (DanielCufino) [reply server/graphql/resolvers/listing.js:8]: The context should contain the JWT token of a user that we can decode and then find a user by to compare against

**Review by DanielCufino — COMMENTED** [review]
looks good!

**Review by DanielCufino — APPROVED** [review]
Great work!

**Pull request merged**